rpcs3/util/atomic.hpp

```diff
@@ -4,10 +4,17 @@
 #include <functional>
 #include <mutex>
 
-#ifdef _M_X64
+#ifndef _MSC_VER
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
```

Comment:
Missing diagnostic pop pragma

Reply:
Correct

Reply:
What about this at the bottom of the file?

```
#ifndef _MSC_VER
#pragma GCC diagnostic pop
#endif
```

Reply:
It's for a different push, you need two pops here

Reply:
Fixed.